tools/mac-distribution/notarize/submit.go

```diff
@@ -0,0 +1,141 @@
+/*
+ *  Copyright 2025 Gravitational, Inc
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package notarize
+
+import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+)
+
+// SubmissionResponseData contains information about the status of a submission.
+//
+// Reference: https://developer.apple.com/documentation/notaryapi/submissionresponse/data-data.dictionary
+type submissionResponseData struct {
+	ID      string `json:"id,omitempty"`
+	Path    string `json:"path,omitempty"`
+	Message string `json:"message,omitempty"`
+	Status  string `json:"status,omitempty"`
+}
+
+// SubmitAndWait is a convenience function that wraps the Submit and WaitForSubmission functions.
+func (t *Tool) SubmitAndWait(pathToPackage string) error {
+	submissionID, err := t.Submit(pathToPackage)
+	if err != nil {
+		return err
+	}
+	if err := t.WaitForSubmission(submissionID); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Submit will submit a package for notarization.
+// A package must be in the format of a zip, package installer (.pkg), or a disk image (.dmg)
+// A success will return a submission ID which can be polled on later.
+func (t *Tool) Submit(pathToPackage string) (id string, err error) {
+	args := []string{
+		"notarytool",
+		"submit", pathToPackage,
+		"--team-id", t.Creds.TeamID,
+		"--apple-id", t.Creds.AppleUsername,
+		"--password", t.Creds.ApplePassword,
```

Comment:
Did you make sure that the command's arguments do not get written to any of the outputs/logs? Can it maybe take the credentials from environment?

Same for wait command below.